File: generated_code/optimized/memory/java/3.java
import java.util.List;

class Solution {
    public boolean belowZero(List<Integer> operations) {
        int balance = 0;
        for (int i = 0, n = operations.size(); i < n; i++) {
            balance += operations.get(i);
            if (balance < 0) {
                return true;
            }
        }
        return false;
    }
}

File: generated_code/optimized/balanced/minimal/cpp/3.cpp
#include <vector>
using namespace std;

bool below_zero(vector<int> operations) {
    long long balance = 0;
    const int* data = operations.data();
    const size_t n = operations.size();

    for (size_t i = 0; i < n; ++i) {
        balance += data[i];
        if (balance < 0) return true;
    }
    return false;
}
